src/store/flow/flow.ts

```diff
@@ -202,6 +202,21 @@ export const mutations: MutationTree<IFlowsState> = {
   flow_removeVendorMetadataByPath(state, {flowId, path}: { flowId: string, path: string }) {
     removeFlowValueByPath(state, flowId, `vendor_metadata.${path}`)
   },
+
+  flow_updateBlockCoordinates(
+      state, {flowId, blockId, coordinates}: { flowId: string, blockId: string, coordinates: {x: number, y: number} },
+  ) {
+    const flow: IFlow = findFlowWith(flowId, state as unknown as IContext)
+    const block: IBlock = findBlockWith(blockId, flow)
+
+    if (!block.ui_metadata) {
+      Vue.set(block, 'ui_metadata', {
+        canvas_coordinates: coordinates,
+      })
+    } else {
+      Vue.set(block.ui_metadata, 'canvas_coordinates', coordinates)
+    }
```

Comment:
A theoretical question @object-required @safydy-r 
What would happen if we do:
```
Vue.set(block, 'ui_metadata', {canvas_coordinates: coordinates})
```
without if-else? I mean what would happen in terms of Vue reactivity if `ui_metadata.canvas_coordinates` exists? What's the danger of vue-setting a higher level object?

I think it's possible it React, for example, right? Even preferrable, I think.
Do you know how Vue handles this? I couldn't find any article on this (maybe I was googling it wrong) - could you help, please?